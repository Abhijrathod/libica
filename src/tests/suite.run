#!/bin/bash

#
# Libica test suite
#

verbosity=$1

out="./suite.out"

testcases=(
"libica_fips_test"

"libica_get_functionlist $verbosity"
"libica_get_version $verbosity"
"icastats_test $verbosity"

"libica_drbg_test $verbosity"

"libica_des_ecb_test $verbosity"
"libica_des_cbc_test $verbosity"
"libica_des_cfb_test $verbosity"
"libica_des_ofb_test $verbosity"
"libica_des_ctr_test $verbosity"

"libica_3des_ecb_test $verbosity"
"libica_3des_cbc_test $verbosity"
"libica_3des_cfb_test $verbosity"
"libica_3des_ofb_test $verbosity"
"libica_3des_ctr_test $verbosity"

"libica_aes128_test $verbosity"
"libica_aes192_test $verbosity"
"libica_aes256_test $verbosity"
"libica_aes_ecb_test $verbosity"
"libica_aes_cbc_test $verbosity"
"libica_aes_cfb_test $verbosity"
"libica_aes_ofb_test $verbosity"
"libica_aes_ctr_test $verbosity"
"libica_aes_gcm_test $verbosity"
"libica_aes_xts_test $verbosity"

"libica_cbccs_test $verbosity"
"libica_ccm_test $verbosity"
"libica_cmac_test $verbosity"

"libica_keygen_test $verbosity 1024 r"
"libica_keygen_test $verbosity 2048 r"
"libica_keygen_test $verbosity 3072 r"
"libica_keygen_test $verbosity 4096 r"
"libica_rsa_key_check_test $verbosity"
"libica_rsa_test $verbosity"

"libica_sha_test/libica_sha_test $verbosity libica_sha_test/sha_test_vectors/*"
)

echo -ne "" &> $out;
for (( i=1; i <= ${#testcases[@]}; i++ ))
do
	echo -ne "Running libica test suite (writing to "$out") ... "$i"/"${#testcases[@]}"\r";
	./${testcases[i-1]} >> $out 2>&1;
done
echo -ne "\n";
