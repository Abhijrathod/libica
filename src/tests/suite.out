All icastats testcases finished successfully
All 3des_cbc testcases finished successfully
All 3des_cfb testcases finished successfully
All 3des_ctr testcases finished successfully
All 3des_ecb testcases finished successfully
All 3des_ofb testcases finished successfully
All AES128 (ECB/CBC) testcases finished successfully
All AES192 (ECB/CBC) testcases finished successfully
All AES256 (ECB/CBC) testcases finished successfully
All AES-CBC testcases finished successfully
All AES-CFB testcases finished successfully
All AES-CTR testcases finished successfully
All AES-ECB testcases finished successfully
All AES-GCM testcases finished successfully.
All AES-OFB testcases finished successfully
All AES-XTS testcases finished successfully
All CBC-CS mode tests finished successfully
All AES-CCM mode tests finished successfully
All CMAC tests finished successfully
All DES-CBC testcases finished successfully
All DES-CFB testcases finished successfully
All DES-CTR testcases finished successfully
All DES-ECB testcases finished successfully
All DES-OFB testcases finished successfully
All ica_get_functionlist tests completed sucessfully
All ica_get_version tests completed sucessfully
All Keygen tests passed successfully
All Keygen tests passed successfully
All Keygen tests passed successfully
All Keygen tests passed successfully
All Keygen tests passed successfully
All RSA testcases finished successfully
All RSA KEY Check testcases finished successfully
All SHA testcases finished successfully
RUNNING SHA-512 DRBG TESTS ... 52 PASSED, 0 FAILED, 52 TOTAL
